### PR TITLE
Button API alignment

### PR DESCRIPTION
**Pull request by langermank**
This PR includes **breaking changes** to bring the PRC Button up to the agreed upon API spec and match implementation with PVC.

- Changes `leadingIcon` and `trailingIcon` to `leadingVisual` and `trailingVisual` (a pattern introduced with ActionList)
- Removes `Button.Counter` as a child component, replacing it with a `count` prop. This change allows us to use the `trailingVisual` slot for counters, and also aligns better with all the other child elements within Button (visuals are props, not child components in this case.)
- Removes the `outline` button variant as we wish to only support `invisible` buttons.

### Screenshots

The only visual change is that the `outline` button variant has been removed.

### Merge checklist

- [x] Added/updated tests
- [x] Added/updated documentation
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

**Comment by langermank** [comment]
If someone wants to pair with me on how to add a codemod to handle all these prop changes, that would be great!

File: .changeset/flat-onions-sort.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"@primer/react": major
+---
+
+- Changes `leadingIcon` and `trailingIcon` to `leadingVisual` and `trailingVisual`
+- Removes `Button.Counter` as a child component, replacing it with a `count` prop. This change allows us to use the `trailingVisual` slot for counters.
```
Comment (broccolinisoup) [inline .changeset/flat-onions-sort.md:6]: Is the prop name `count` or `trailingVisualCount`? 🤔

File: docs/content/Button.mdx [hunk 2]
```diff
@@ -98,17 +90,12 @@ A separate component called `IconButton` is used if the action shows only an ico
 </>
 ```
 
-### Counter component
+### Button with counter
 
-A common use case for primer is a button with a counter component which shows the child count value.
-We provide `Button.Counter` as a composite component which requires you to provide a number as child.
-The counter will match the `variant` styles of the parent button.
+To show a count value as a trailing visual inside `Button`, pass a value to the `count` prop. The counter will match the `variant` styles of the parent button.
```
Comment (broccolinisoup) [inline docs/content/Button.mdx:95]: Same comment as above (prop name)

**Review by joshblack — APPROVED** [review]
🥳

**Pull request merged**